test/model/content_matchers.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:test/test.dart';
+import 'package:zulip/model/content.dart';
+
+abstract class ContentNodeMatcher extends Matcher {
+}
+
+class ZulipContentMatcher extends ContentNodeMatcher {
+  ZulipContentMatcher(this.nodes);
+
+  final List<Matcher> nodes;
+
+  @override
+  bool matches(covariant ZulipContent item, Map<dynamic, dynamic> matchState) {
+    // TODO surely this is common boilerplate we can avoid, right?
+    //   ... Well, doesn't seem like the matcher package has a way:
+    //     https://pub.dev/documentation/matcher/latest/matcher/matcher-library.html
+    //   Maybe try its proposed successor which is in beta?:
+    //     https://pub.dev/packages/checks
+    if (item.nodes.length != nodes.length) return false;
+    for (var i = 0; i < nodes.length; i++) {
+      if (!nodes[i].matches(item.nodes[i], matchState)) {
+        return false;
+      }
+    }
```

Comment:
> ```dart
> /// If [expected] is a [Matcher], then it matches using that. Otherwise it tests
> /// for equality using `==` on the expected value.
> ```

Yeah — I was fooled by the name and summary, which seemed awfully clear that it meant *equal* and not recursively applying general matchers.

It turns out the `checks` package has a similar story, but perhaps there's an opportunity to improve it there before it's stable.